src/scripts/sniperino.ts

```diff
@@ -0,0 +1,101 @@
+// Description:
+//  Beat the sniperino
+//  
+//
+// Commands:
+//  sniperino( me)? - start a new game of sniperino 
+//  snipe( me)? - roll a snipe roll in an attempt to beat the sniperino
+//
+// Author:
+//  Steve Shipsey
+
+import { Robot } from "tsbot";
+
+interface ISniper {
+    name: string;
+    gamesPlayed: number;
+    gamesWon: number;
+    currentSnipe?: number;
+
+    getWinRate(): number;
+    getNewSnipe();
+}
+
+class Sniper implements ISniper {
+    public name: string;
+    public gamesPlayed: number;
+    public gamesWon: number;
+    public currentSnipe: number;
+    
+    constructor(n: string) {
+        this.name = n.slice(0, n.indexOf(" "));
+        this.gamesPlayed = 0;
+        this.gamesWon = 0;
+        //this.currentSnipe = this.getNewSnipe();
+    }
+
+    public getWinRate(): number {
+        return this.gamesPlayed > 0 ? (this.gamesWon / this.gamesPlayed) * 100 : 0;
+    }
+
+    public getNewSnipe() {
+        this.currentSnipe = Math.floor(Math.random() * 99) + 1;
+    }
+}
+
+let game = (robot: Robot) => {
+
+    // Get a sniper with a given name. If one doesn't exist, create it.
+    let getSniper = (name: string) => {
+    
+        // Check for active sniper, if not, create one
+        if (robot.brain.get(name) == null) {
```

Comment:
It might be interesting to separate the currentSnipe from the player's overall stats. Since the stats are what you want persisted "long-term" you could use the brain for those. And the currentSnipe could just be some local dictionary.

Sitenote: I don't remember if we have a functional brain configured. My guess is that we do not since I dont have any persistent data volumes attached to the docker container.